web/src/pages/insights/SummaryTable.tsx

```diff
@@ -0,0 +1,294 @@
+            className={rowClassName}
+        >
+            {summaryTableHeaderCellConfigs.map((config) =>
+                getSummaryCell(summary, config.key, isDarkMode)
+            )}
+        </SUITable.Row>
+    )
+}
+
+const SummaryTable: React.FC<SummaryTableProps> = ({
+    filteredData,
+    getUniqueOptionsForColumn,
+    handleSelectionChange,
+    getSelectedOptionsForColumn,
+}) => {
+    const [sortColumns, setSortColumns] = useState<
+        Array<{ column: ColumnKey; direction: 'ascending' | 'descending' }>
+    >([])
+    const handleSort = (column: ColumnKey, isMultiSort: boolean) => {
+        if (isMultiSort) {
+            const existingColumnIndex = sortColumns.findIndex(
+                (sortColumn) => sortColumn.column === column
+            )
+
+            if (existingColumnIndex !== -1) {
+                const updatedSortColumns = [...sortColumns]
+                updatedSortColumns[existingColumnIndex].direction =
+                    updatedSortColumns[existingColumnIndex].direction === 'ascending'
+                        ? 'descending'
+                        : 'ascending'
+                setSortColumns(updatedSortColumns)
+            } else {
+                setSortColumns([...sortColumns, { column, direction: 'ascending' }])
+            }
+        } else {
+            if (sortColumns.length === 1 && sortColumns[0].column === column) {
+                setSortColumns([
+                    {
+                        column,
+                        direction:
+                            sortColumns[0].direction === 'ascending' ? 'descending' : 'ascending',
+                    },
+                ])
+            } else {
+                setSortColumns([{ column, direction: 'ascending' }])
+            }
+        }
+    }
+
+    const sortedData = React.useMemo(() => {
+        const data = [...filteredData]
+        data.sort((a, b) => {
+            for (const { column, direction } of sortColumns) {
+                const valueA = a[column as keyof ProjectInsightsSummary]
+                const valueB = b[column as keyof ProjectInsightsSummary]
+                if (valueA === valueB) continue
+                if (typeof valueA === 'number' && typeof valueB === 'number') {
+                    return direction === 'ascending' ? valueA - valueB : valueB - valueA
+                } else {
+                    return direction === 'ascending'
+                        ? String(valueA).localeCompare(String(valueB))
+                        : String(valueB).localeCompare(String(valueA))
+                }
+            }
+            return 0
+        })
+        return data
+    }, [filteredData, sortColumns])
+
+    return (
+        <div>
+            <Table sortable>
+                <SUITable.Header>
+                    <SUITable.Row>
+                        {summaryTableHeaderCellConfigs.map((config) => (
+                            <HeaderCell
+                                key={config.key}
```

Comment:
EG: here you do need the component key, and it's neighbourly unique which is perf 👌